src/Writer.zig

```diff
@@ -183,6 +183,17 @@ pub fn text(writer: *Writer, s: []const u8) anyerror!void {
     writer.state = .text;
 }
 
+// insert some existing XML document without escaping anything
+pub fn embed(writer: *Writer, s: []const u8) anyerror!void {
+    switch (writer.state) {
+        .after_structure_end, .text => {},
+        .element_start => try writer.raw(">"),
+        .start, .after_bom, .after_xml_declaration, .end => unreachable,
+    }
```

Comment:
```suggestion
    switch (writer.state) {
        .start, .after_bom, .after_xml_declaration, .after_structure_end, .text, .end => {},
        .element_start => try writer.raw(">"),
    }
```
I don't see any reason to restrict this function to operate only inside the root element: it's reasonable that someone might want to use it to embed a DTD at the beginning of the document, for example, which isn't currently possible with the other methods.

Reply:
That's a good point, I was thinking too narrowly about my own use case, embedding xml fragments inside an existing document. Fixed in https://github.com/ianprime0509/zig-xml/pull/39/commits/f54c606ce886ac26d6dc0e902733c1aa71a5013e